sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -746,6 +750,270 @@ case class DescribeTableCommand(
   }
 }
 
+/**
+ * Command that looks like
+ * {{{
+ *   DESCRIBE [EXTENDED|FORMATTED] table_name partitionSpec? [AS JSON];
+ * }}}
+ */
+case class DescribeTableJsonCommand(
+   table: TableIdentifier,
+   partitionSpec: TablePartitionSpec,
+   isExtended: Boolean) extends LeafRunnableCommand {
+  override val output = DescribeCommandSchema.describeJsonTableAttributes()
+
+  override def run(sparkSession: SparkSession): Seq[Row] = {
+    val jsonMap = mutable.LinkedHashMap[String, JValue]()
+    val catalog = sparkSession.sessionState.catalog
+
+    if (catalog.isTempView(table)) {
+      if (partitionSpec.nonEmpty) {
+        throw QueryCompilationErrors.descPartitionNotAllowedOnTempView(table.identifier)
+      }
+      val schema = catalog.getTempViewOrPermanentTableMetadata(table).schema
+      describeColsJson(schema, jsonMap, header = false)
+    } else {
+      val metadata = catalog.getTableRawMetadata(table)
+      val schema = if (metadata.schema.isEmpty) {
+        // In older versions of Spark,
+        // the table schema can be empty and should be inferred at runtime.
+        sparkSession.table(metadata.identifier).schema
+      } else {
+        metadata.schema
+      }
+
+      addKeyValueToMap("table_name", JString(metadata.identifier.table), jsonMap)
+      table.catalog.foreach(catalog => addKeyValueToMap("catalog_name", JString(catalog), jsonMap))
+      table.database.foreach { db =>
+        addKeyValueToMap("namespace", JArray(List(JString(db))), jsonMap)
+        addKeyValueToMap("schema_name", JString(db), jsonMap)
+      }
+
+      describeColsJson(schema, jsonMap, header = false)
+      describeClusteringInfoJson(metadata, jsonMap)
+
+      if (partitionSpec.nonEmpty) {
+        // Outputs the partition-specific info for the DDL command:
+        // "DESCRIBE [EXTENDED|FORMATTED] table_name PARTITION (partitionVal*)"
+        describePartitionInfoJson(sparkSession, catalog, metadata, jsonMap)
+      } else {
+        describeFormattedTableInfoJson(metadata, jsonMap)
+      }
+    }
+
+    Seq(Row(compact(render(JObject(jsonMap.toList)))))
+  }
+
+  private def addKeyValueToMap(
+      key: String,
+      value: JValue,
+      jsonMap: mutable.LinkedHashMap[String, JValue]): Unit = {
+    print(s"\nAdding key: $key, value: $value to jsonMap")
+    // Already added to jsonMap in DescribeTableJsonCommand
+    val excluded_keys = Set("catalog", "schema", "database", "table")
```

Comment:
```suggestion
    val excludedKeys = Set("catalog", "schema", "database", "table")
```

Reply:
shall we make it a class member? To avoid instantiating it for every `addKeyValueToMap` invocation.